### PR TITLE
Initial Overview

### PR DESCRIPTION
**Pull request by binji**

### CODE REVIEW

File: proposals/memory64/Overview.md [hunk 1]
```diff
@@ -2,12 +2,188 @@
...
 ## Overview
 
-TODO
+### Structure
+
+* The [limits][syntax limits] structure is changed to use `u64`
+  - `limits ::= {min u64, max u64?}`
+
+* The [memory type][syntax memtype] structure is extended to have an index type
+  - `memtype ::= limits valtype`
```
Comment (rossberg) [inline proposals/memory64/Overview.md]: Do we really want to use valtype here? Maybe it's better to define a proper sub category `idxtype ::= i32 | i64`. It seems weird to syntactically allow floats or even references.

File: proposals/memory64/Overview.md [hunk 2]
```diff
@@ -2,12 +2,188 @@
...
+  - `t.load memarg`
+  - `t.loadN_sx  memarg`
+  - `t.store memarg`
+  - `t.storeN memarg`
+  - `memory.size`
+  - `memory.grow`
+  - (spec text omitted)
+
+* [memory.grow][exec memgrow] is changed to check for a size greater than
+  2<sup>64</sup> - 1, and to return 2<sup>64</sup>-1 when `memory.grow` fails.
```
Comment (rossberg) [inline proposals/memory64/Overview.md]: The value would now depend on the index type, wouldn't it?

**Review by rossberg — COMMENTED** [review]
Makes sense to me.

File: proposals/memory64/Overview.md (hunk 1 above)
Comment (binji) [reply proposals/memory64/Overview.md]: Works for me, I'll update.

File: proposals/memory64/Overview.md (hunk 2 above)
Comment (binji) [reply proposals/memory64/Overview.md]: Yep, fixed.

**Pull request merged**